server_test.go

```diff
@@ -134,3 +134,25 @@ func TestServer_ServeHTTP_health(t *testing.T) {
 	assert.Equal(t, 200, w.Code)
 	assert.Equal(t, "OK\n", w.Body.String())
 }
+
+func BenchmarkServe(b *testing.B) {
+	p := new(pub)
+
+	s := Server{
+		Log:       log.New(ioutil.Discard, "", log.LstdFlags),
+		Topic:     "builds",
+		Publisher: p,
+	}
+
+	r, err := http.NewRequest("POST", "/build", bytes.NewBufferString(`{ "foo": "bar" }`))
+	if err != nil {
+		b.Error(err)
+	}
+	r.Header.Set("Content-Type", "application/json")
+
+	w := httptest.NewRecorder()
```

Comment:
To help isolate your benchmark and reduce the noise as much as possible, how about implementing a dummy ResponseWriter here instead of using the test recorder?